tests/functional/test_libraries.py

```diff
@@ -47,6 +47,76 @@ def test_tkinter(pyi_builder):
     pyi_builder.test_script('pyi_lib_tkinter.py')
 
 
+# In contrast to test_tkinter, which performs basic import test and verifies that the environment variables are properly
+# set, this is a full functional test; we try to create a Tk window with label and button, and register a timer to shut
+# down the application. Doing so verifies that all Tcl/Tk files (e.g., .tcl scripts from library directories) are
+# properly collected.
+#
+# The prerequisite for this test is that tkinter can be used unfrozen, so try instantiating a window in a subprocess
+# to verify that this is the case. This check should cover the following scenarios:
+#  - tkinter missing
+#  - import of tkinter crashes python interpreter
+#  - tkinter.Tk() fails due to DISPLAY not being set on linux
+#  - tkinter.Tk() fails due to faulty build (e.g., due to Tcl/Tk version mix-up, as seen with python <= 3.10 builds on
+#    macos-12 GHA runners; https://github.com/actions/setup-python/issues/649#issuecomment-1745056485)
+def _tkinter_fully_usable():
+    from PyInstaller import isolated
+
+    @isolated.decorate
+    def _create_tkinter_window():
+        import tkinter
+        tkinter.Tk()
+
+    try:
+        _create_tkinter_window()
+    except Exception:
+        return False
+
+    return True
+
+
+@pytest.mark.skipif(not _tkinter_fully_usable(), reason="tkinter is not fully usable.")
```

Comment:
Can we move this check into the test (`if not _tkinter_fully_usable(): pytest.skip(...)`)? Adding non trivial `skipif()` checks increases pytest's startup time which I find disproportionately irksome.